File: backend/nestjs/src/models/timeline/post.model.ts
export class ReactionModel {
  uid: string;
  userUid: string;
  updatedAt: string;

  constructor(
    { uid, userUid, updatedAt }:
      {
        uid: string;
        userUid: string;
        updatedAt: string;
      },
  ) {
    Object.assign(this, {
      uid, userUid, updatedAt
    });
  }
}

export class CommentaryModel {
  uid: string;
  userUid: string;
  updatedAt: string;
  commentary: string;

  constructor(
    { uid, userUid, updatedAt, commentary }:
      {
        uid: string;
        userUid: string;
        updatedAt: string;
        commentary: string;
      },
  ) {
    Object.assign(this, {
      uid, userUid, updatedAt, commentary
    });
  }
}

export class PostModel {
  uid: string;
  userUid: string;
  userPicture: string;
  userName: string;
  title: string;
  body: string;
  pictures: Array<string>;
  commentarys: Array<CommentaryModel>;
  createdAt: string;
  updatedAt: string;


  constructor(
    { uid, userUid, userPicture, userName, title, body, pictures, commentarys, createdAt, updatedAt }:
      {
        uid: string;
        userUid: string;
        userPicture: string;
        userName: string;
        title: string;
        body: string;
        pictures: Array<string>;
        commentarys: Array<CommentaryModel>;
        createdAt: string;
        updatedAt: string;
      },
  ) {
    Object.assign(this, {
      uid, userUid, userPicture, userName, title, body, pictures, commentarys, createdAt, updatedAt
    });
  }
}

File: app/lib/core/models/post/post_model.dart
import 'package:beautybook/core/models/base_model.dart';
import 'package:json_annotation/json_annotation.dart';

part 'post_model.g.dart';

@JsonSerializable(nullable: true, anyMap: true)
class PostModel extends BaseModel<String> {
  final String uid;
  final String userUid;
  final String userName;
  final String userPicture;
  final String title;
  final String body;
  final List<String> pictures;
  final DateTime createdAt;
  final DateTime updatedAt;

  PostModel({
    this.uid,
    this.userUid,
    this.userPicture,
    this.userName,
    this.title,
    this.body,
    this.pictures,
    this.createdAt,
    this.updatedAt,
  });

  factory PostModel.fromJson(Map json) => _$PostModelFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PostModelToJson(this);

  @override
  String get id => this.uid;

  @override
  List<Object> get props => [uid];
}
